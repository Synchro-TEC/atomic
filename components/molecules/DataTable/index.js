import React from 'react';
import PropTypes from 'prop-types';
import nanoid from 'nanoid';
import _isFunction from 'lodash/isFunction';

const DataTable = ({ cols, data }) => {
  return (
    <table className="sv-table with--borders with--hover">
      <thead>
        <tr>
          {cols.map(col => (
            <th key={nanoid()}>{col.label}</th>
          ))}
        </tr>
      </thead>
      <tbody>
        {data.map(d => (
          <tr key={nanoid()}>
            {cols.map(col => (
              <td key={nanoid()}>{_isFunction(col.accessor) ? col.accessor(d) : d[col.accessor]}</td>
            ))}
          </tr>
        ))}
      </tbody>
    </table>
  );
};

DataTable.propTypes = {
  cols: PropTypes.array.isRequired,
  data: PropTypes.array.isRequired,
};

export default DataTable;
