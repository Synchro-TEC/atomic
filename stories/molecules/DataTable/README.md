The `<DataTable>` component is a button.