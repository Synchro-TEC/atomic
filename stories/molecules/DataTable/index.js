import React from 'react';
// import { text, boolean, object } from "@storybook/addon-knobs";
// import { action } from '@storybook/addon-actions';
// import { optionalSelect } from '../../../utils/optionalSelect';
import { State, Store } from '@sambego/storybook-state';
import readme from './README.md';
import DataTable from '../../../components/molecules/DataTable';

const store = new Store({
  cols: [{ label: 'Nome', accessor: 'name' }, { label: 'E-mail', accessor: 'email' }],
  data: [
    { name: 'Fulano', email: 'Fulano de Tal' },
    { name: 'Fulano', email: 'Fulano de Tal' },
    { name: 'Fulano', email: 'Fulano de Tal' },
    { name: 'Fulano', email: 'Fulano de Tal' },
    { name: 'Fulano', email: 'Fulano de Tal' },
    { name: 'Fulano', email: 'Fulano de Tal' },
    { name: 'Fulano', email: 'Fulano de Tal' },
  ],
});

const component = () => {
  return (
    <State store={store}>
      <DataTable cols={store.cols} data={store.data} />
    </State>
  );
};

export default [readme, component];
