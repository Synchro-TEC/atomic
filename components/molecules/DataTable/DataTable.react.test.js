import React from 'react';
import DataTable from './index';

describe('DataTable Component', () => {
  test('Render DataTable Component', () => {
    const datatable = shallow(<DataTable cols={[]} data={[]} />);
    expect(datatable).toMatchSnapshot();
  });
});
