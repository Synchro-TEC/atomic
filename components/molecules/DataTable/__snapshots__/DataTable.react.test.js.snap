// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable Component Render DataTable Component 1`] = `
<table
  className="sv-table with--borders with--hover"
>
  <thead>
    <tr />
  </thead>
  <tbody />
</table>
`;
